scripts/favourite_meal_notifier.ts

```diff
@@ -0,0 +1,171 @@
+        // Expired token
+        tokensToDelete.add(sub.device.id);
+        return false;
+      }
+      // Token is not expired or we don't know about the expiry
+      return true;
+    })
+    .map((sub): FBDebugMessage => {
+      tokensToRefresh.add(sub.device.id);
+      return {
+        deviceId: sub.device.id,
+        message: {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          token: sub.device.registrationToken!,
+          data: {
+            mealId: mealIdString,
+          },
+        },
+      };
+    });
+
+  if (validMessages.length === 0) {
+    logger.info("No registration tokens found. Exiting early.");
+    return;
+  }
+  logger.info(`Found ${validMessages.length} subscribed device(s)`);
+  // Init FB
+  try {
+    if (!getApps().length) {
+      initializeApp({
+        credential: applicationDefault(),
+      });
+      logger.info("Firebase app initialized.");
+    }
+  } catch (error) {
+    logger.error(
+      "Failed to initialize Firebase app. Exiting early. Error: ",
+      error,
+    );
+    return;
+  }
+  const messagingService = getMessaging();
+  let successCount = 0;
+
+  for (const message of validMessages) {
+    try {
+      await messagingService.send(message.message);
+      logger.debug(`Device ${message.deviceId}: Success`);
+      successCount++;
+    } catch (error) {
+      const fbError = error as { code: string; message: string };
+      if (fbError.code === "messaging/registration-token-not-registered") {
+        // Token was not expired in our DB but FB deemed it expired
+        logger.debug(
+          `Device ${message.deviceId}: Failure - Token expired or not registered in FB`,
+        );
+        tokensToDelete.add(message.deviceId);
+        tokensToRefresh.delete(message.deviceId);
+      } else if (fbError.code === "messaging/invalid-argument") {
+        // Token is not valid because it is simply not - not due to expiry
+        logger.debug(
+          `Device ${message.deviceId}: Failure - Token is not valid`,
+        );
+        tokensToDelete.add(message.deviceId);
+        tokensToRefresh.delete(message.deviceId);
+      } else {
+        // Error not related to the token itself
+        logger.warn(
+          `Device ${message.deviceId}: Failure - unknown error ${fbError.code}. Error: ${fbError.message}`,
+        );
+      }
+    }
+  }
+  logger.info(
+    `Successfully sent ${successCount} out of possible ${validMessages.length} messages.`,
+  );
+  await updateTokenState(tokensToDelete, tokensToRefresh);
+}
+
+async function updateTokenState(
+  tokensToDelete: Set<number>,
+  tokensToRefresh: Set<number>,
+) {
+  logger.info("Updating token state in the database...");
+  if (tokensToRefresh.size > 0) {
+    try {
+      await Device.refreshTokenTimestamps([...tokensToRefresh]);
+      logger.info(`Refreshed ${tokensToRefresh.size} tokens`);
+    } catch (error) {
+      logger.warn(`Failed to refresh tokens: `, error);
+    }
```

Comment:
this isn't really "refreshing", but rather just marking it as refreshed in the db